rllib/agents/ppo/ppo_torch_policy.py

```diff
@@ -161,11 +161,38 @@ def vf_preds_and_logits_fetches(policy, input_dict, state_batches, model,
                                 action_dist):
     """Adds value function and logits outputs to experience train_batches."""
     return {
-        SampleBatch.VF_PREDS: policy.model.value_function().cpu().numpy(),
-        BEHAVIOUR_LOGITS: policy.model.last_output().cpu().numpy(),
+        SampleBatch.VF_PREDS: policy.model.value_function().numpy(),
+        BEHAVIOUR_LOGITS: policy.model.last_output().numpy(),
     }
 
 
+def postprocess_ppo_gae(policy,
+                        sample_batch,
+                        other_agent_batches=None,
+                        episode=None):
+    """Adds the policy logits, VF preds, and advantages to the trajectory."""
+    # Do all post-processing always with no_grad().
+    # Not using this here will be a
```

Comment:
"will be a ____" ... memory leak?

Reply:
yes :) . I'll add a comment everywhere, we do this.

Reply:
done